### ISSUE TITLE
Cleanup

### ISSUE THREAD

**Issue by MggMuggins** [redox-os/userutils#31]
A few things I thought I should document in an issue:

Wondering if I should remove arg parsing from `login` completely, since there is only `--help`, and it only prints the man page right now. Would simplify that significantly.

I'm also thinking about dropping `sudo` from this crate, for a few reasons:
- `sudo` is a bigger program than many think, and implementing sudo correctly takes a lot of effort. You need to have the configuration file, and some command line options, probably sudoedit, etc.
- For packaging reasons: many os's package sudo separately from other utilities or coreutils.

rudo might be an alternative, although it's PAM based so we'd either need to impl PAM for redox or port PAM, neither of which I'd be thrilled about doing, or building a different backend into rudo for redox_users. @shawnanastasio I guess I should mention you on this one since rudo is your program...

**Comment by shawnanastasio** [comment]
It would be easy enough to add a redox_users backend to rudo, and I'd be willing to do it if there was any interest.

Aside from that, I had also planned on improving rudo in a number of ways including a more intuitive configuration file format, as well as some security improvements. I had forgone making these changes due to a lack of perceived interest, but I'd be happy to pick it back up again if that were to change.

**Comment by MggMuggins** [comment]
That's awesome!!! If you wanted to try doing a redox backend, this is the API I've been working on: https://github.com/redox-os/users

There's no pressure to work on it, so feel free to not pay us any mind if you don't feel like doing that. We'd super appreciate it if you'd be willing to maintain it for us though!

Something I would like to mention:
One thing to keep in mind is how easy it is to screw up your configuration files. For example, `/etc/passwd` on most unixes is extremely simple, and parsing is easy. Moreover, if one line of the file is invalid, it does not invalidate the rest of the file, it just does not parse that user.

I would say that the sudoers file should likewise be as difficult to invalidate with syntax errors as possible, that way sudo will still work even if parts of the file are not valid. Just something I thought I'd mention, since it seems like a good thing to bear in mind when designing things like that.

**Comment by shawnanastasio** [comment]
Awesome, I'll take a look and create a redox branch of rudo for testing in the near future.

As for the configuration file, I'm trying to avoid the pitfalls of custom configuration file formats by using an existing standard with an established parsing library (currently JSON, looking at switching to XML or TOML). In many cases, the parsing libraries are quite strict and will refuse to continue if there's a syntax error. I was planning on getting around this by using an approach similar to sudo's, by having a custom edit command that wraps your system text editor and validates the syntax on save to prevent you from writing a broken file.

**Comment by MggMuggins** [comment]
Yes, even with a much more forgiving format, something like `visudo` Would be really nice. I think TOML would make a good choice, because it seems to be more forgiving than JSON or XML. Also it's concise and fairly easy to understand when doing configurations.

**Comment by MggMuggins** [comment]
I went ahead and ported login in 69838054ab28d9b623838c7452aead63edaa4a61.
